### PR TITLE
Fix(Scripts/BlackrockSpire): rewrite Mor Grayhoof fight

### LINKED ISSUE #9111: Boss of D2 set questline Mor Grayhoof issues
### What client do you play on?

enUS

### Faction

- [X] Alliance
- [X] Horde

### Content Phase:

- [ ] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [X] 50-59

### Current Behaviour

[Mor Grayhoof](https://wowgaming.altervista.org/aowow/?npc=16080) summoned by [Brazier of Beckoning](https://wowgaming.altervista.org/aowow/?item=22049). Current state:
1. After summon Mor Grayhoof is instantly start attacking random player in 50y range
2. Despawn timer of Mor Grayhoof about 30 seconds
3. Mor Grayhoof doesn't have yells
4. Mor Grayhoof has incorrect ability scripting

https://user-images.githubusercontent.com/85256266/141116791-429d635d-e3bf-4544-864a-5c1fee5d1d20.mp4

### Expected Blizzlike Behaviour

1. Mor Grayhoof should not attack after summon. Only if someone in agro range.
2. Mor Grayhoof remain until defeated. There is no need to resummon him upon wiping, or even if the brazier holder leaves the instance or group. Only one Kormok per instance.
3. Mor Grayhoof should have on summon and on death yell
4. Mor Grayhoof missing in human form: 
- Hurricane
- AoE Moonfire
- Healing Touch
- Chain Light?

Mor Grayhoof missing in Spirit Darter from form: 
- Sleep
- Arcane Explosion
- Reflection

Mor Grayhoof forms is looks like random exept Spirit Darter from 25%

### Source

Wowwiki [Mor Grayhoof](https://wowwiki-archive.fandom.com/wiki/Mor_Grayhoof?oldid=2295320) has good ability guide

Wowhead [Mor Grayhoof](https://www.wowhead.com/npc=16080/mor-grayhoof)

https://youtu.be/myINuTQ_t0Y?t=14613
https://youtu.be/YZWU0-t8GvU?t=132

### Steps to reproduce the problem

.go c id 9237 - teleport to War Master Voone
.cast 27184

### Extra Notes

_No response_

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/12e92b58082dfda8b73b50aa97b98b9479bb7629

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

### PR DESCRIPTION
**Pull request by Nefertumm**
## Changes Proposed:
-  Move SAI to cpp.
-  Added missing spells from different phases.

## Issues Addressed:

- Closes #9111

## Tests Performed:

-  Ran the sql, built, tested in-game.

## How to Test the Changes:

1.  .go c id 9237 - teleport to War Master Voone
2.  .cast 27184

## Known Issues and TODO List:

- [ ]  Find the proper sleep spell.
- [ ]  Timers?

### CODE REVIEW

**Comment by IntelligentQuantum** [comment]
Tested

**Review by IntelligentQuantum — APPROVED** [review]

File: src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_mor_grayhoof.cpp [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+
+class boss_mor_grayhoof : public CreatureScript
+{
+public:
+    boss_mor_grayhoof() : CreatureScript("boss_mor_grayhoof") { }
+
+    struct boss_mor_grayhoofAI : public BossAI
+    {
+        boss_mor_grayhoofAI(Creature* creature) : BossAI(creature, DATA_MOR_GRAYHOOF)
+        {
```
Comment (Kitzunu) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_mor_grayhoof.cpp:81]: No. Use new script model

File: src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_mor_grayhoof.cpp [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+    CreatureAI* GetAI(Creature* creature) const override
+    {
+        return GetBlackrockSpireAI<boss_mor_grayhoofAI>(creature);
+    }
+};
+
+void AddSC_boss_mor_grayhoof()
+{
+    new boss_mor_grayhoof();
+}
```
Comment (Kitzunu) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_mor_grayhoof.cpp:243]: .

**Review by Kitzunu — CHANGES_REQUESTED** [review]

File: src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_mor_grayhoof.cpp (hunk 1 above)
Comment (Kitzunu) [reply src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_mor_grayhoof.cpp:81]: https://github.com/azerothcore/azerothcore-wotlk/commit/430fa147fd340223400f6df968d0726510bb1c99

**Comment by Nefertumm** [comment]
@Kitzunu

**Review by Kitzunu — APPROVED** [review]

**Pull request merged**